src/test/java/teammates/test/pageobjects/GoogleLoginPage.java

```diff
@@ -84,24 +89,21 @@ private void handleApprovalPageIfAny() {
         waitForRedirectIfAny();
         boolean isPageRequestingAccessApproval = getPageSource().contains(EXPECTED_SNIPPET_APPROVAL);
         if (isPageRequestingAccessApproval) {
-            click(By.id("persist_checkbox"));
+            markCheckBoxAsChecked(browser.driver.findElement(By.id("persist_checkbox")));
             click(By.id("approve_button"));
             waitForPageToLoad();
         }
     }
 
     private void waitForRedirectIfAny() {
-        By metaRefreshBy = By.cssSelector("meta[http-equiv='refresh']");
-        if (isElementPresent(metaRefreshBy)) {
-            waitForElementToDisappear(metaRefreshBy);
-            waitForPageToLoad();
-        }
-
-        By noScriptBy = By.tagName("noscript");
-        if (isElementPresent(noScriptBy)) {
-            waitForTextContainedInElementAbsence(noScriptBy, "&lt;meta http-equiv=\"refresh\"");
-            waitForPageToLoad();
-        }
+        String loginRedirectUrl = Config.APP_URL + "/_";
+        WebDriverWait wait = new WebDriverWait(browser.driver, TestProperties.TEST_TIMEOUT);
+        wait.until((WebDriver d) -> {
```

Comment:
Just wanna say I like how I can finally see lambda arrow in TEAMMATES code base 😛